### ISSUE TITLE
Mention exclusions in the `constrain.Rigid` documentation.

### ISSUE THREAD

**Issue by joaander** [glotzerlab/hoomd-blue#1465]
### Description

Mention in the `constrain.Rigid` that users should enable the `'body'` exclusion in the neighbor list.

### Motivation and context

Body exclusions prevent the computation of self-interactions. These self-interactions may lead to:
* Unexpected potential energy totals.
* Particles thrown out of the box when the self-interaction leads to very large forces.
* A reduction in momentum/energy conservation when the self-interaction leads to moderate forces.

**Comment by TonyBhargav** [comment]
hi @joaander  , I would like to work on this documentation as a part of academic course work ,It would be a great opportunity for me to learn about the project architecture and design as well. Could you please assign it to me

**Comment by joaander** [comment]
@TonyBhargav HOOMD-blue is an open-source project. We will review pull requests whether you are assigned to an issue or not. We typically do not assign issues to individuals that are not part of the core development team.

Please review the [contribution](https://hoomd-blue.readthedocs.io/en/v3.8.0/contributing.html) and the [code style](https://hoomd-blue.readthedocs.io/en/v3.8.0/style.html) guides before starting your work. Ask here if you have any questions.
